tests/public-api/examples/optimization/solver.py

```diff
@@ -0,0 +1,194 @@
+    """
+    Call a solver on osparc
+    """
+    if cfg is None:
+        cfg = get_config()
+    assert input_file.name.endswith('.h5'), f'{str(input_file)} must be a h5 file'
+    try:
+        with osparc.ApiClient(cfg) as api_client:
+            # APIs
+            users_api = osparc.UsersApi(api_client)
+            files_api = FilesApi(api_client)
+            solvers_api = SolversApi(api_client)
+            users_api.get_my_profile() # validate access
+
+            # create objects
+            input: File = files_api.upload_file(file=input_file)
+            solver: Solver = solvers_api.get_solver_release(solver_key, solver_version)
+            job: Job = solvers_api.create_job( solver.id, solver.version, JobInputs({"input_1": input}))
+            logging.info(f'Done creating job using input file: {input_file}')
+
+            # solve
+            logging.info(f'Start solving job: {job.id}')
+            status: JobStatus = solvers_api.start_job(solver.id, solver.version, job.id)
+            with tqdm(total=100) as bar:
+                while not status.stopped_at:
+                    time.sleep(0.5)
+                    status = solvers_api.inspect_job(solver.id, solver.version, job.id)
+                    bar.update(status.progress)
+            if status.state != 'SUCCESS':
+                logging.error(f'Failed job {job.id} with status {status.state}')
+                log: List[str] = [f'Failed to solve job with status {status.state}', 'Server log:']
+                log.append( generate_isolve_log(solvers_api, solver_key, solver_version, job.id) )
+                raise Exception('\n'.join(log))
+            logging.info(f'Successfully solved job: {job.id}')
+
+            # download outputs
+            outputs: JobOutputs = solvers_api.get_job_outputs(solver.id, solver.version, job.id)
+            for _, result in outputs.results.items():
+                file = files_api.download_file(result.id)
+                if result.filename == 'output.h5':
+                    shutil.move(file, output_file)
+                elif result.filename == 'log.tgz':
+                    shutil.move(file, log_path)
+                else:
+                    logging.info(f'Received unexpected output: {result.filename}')
+                    continue
+                logging.info(f'Successfully downloaded {result.filename}')
+
+    except osparc.exceptions.ApiException as e:
+        api_client.close()
+        expt = json.loads(e.body)
+        raise Exception('\n'.join(expt["errors"])) from None
+
+
+if __name__ == '__main__':
+    cfg = osparc.Configuration(
+        username="1c9034e8-713c-5bec-b0ce-6aa070e1b329",
+        password="a1724945-1f91-5dca-8a0c-8efb018028b0"
+    )
+    solver_key: str = "simcore/services/comp/isolve"
+    solver_version: str = "2.1.16"
+    project: Path = Path(os.getcwd()) / 'project'
```

Comment:
MINOR: `Path("./project)` will already refer to the CWD but if you want to be even more explicit you can use https://docs.python.org/3/library/pathlib.html#pathlib.Path.cwd

Reply:
Yep, I have removed this `if __name__ == '__main__'` statement completely as it is not really relevant. The main thing which should be run is the `example.py` file.